File: SignalR/SignalR.Selfhosted/Messages/Models/ConversationModel.cs
using SignalR.SelfHosted.Users.Models;
using System;

namespace SignalR.SelfHosted.Messages.Models;

public class ConversationModel
{
    public int Id { get; set; }

    public string Title { get; set; }

    public UserModel CreatorUser { get; set; }

    public DateTime CreatedAt { get; set; }
}

File: SignalR.Client/src/app/models/ConversationModel.ts
import { UserModel } from './UserModel';

export interface ConversationModel {
  id: number;
  title: string;
  creatorUser?: UserModel;
  createdAt: Date;
}

export interface ConversationCreateModel {
  title: string;
  creatorUserId: number;
}

export interface ConversationAudienceModel extends ConversationModel {
  audienceUsers: UserModel[];
}
